app/models/user.server.ts

```diff
@@ -3,21 +3,29 @@ import logger from '~/lib/logger.server';
 import { buildUserBaseDomain } from '~/utils';
 
 import type { User as PrismaUser } from '@prisma/client';
+import { getEffectiveName } from '~/session.server';
 
 export interface User extends PrismaUser {
   // the base domain to use for all DNS records created by this user
   // (e.g., jsmith.starchart.com with records using *.jsmith.starchart.com)
   baseDomain: string;
+  effectiveUsername: string;
+  isAdmin: boolean;
 }
 
-export async function getUserByUsername(username: PrismaUser['username']) {
+export async function getUserByUsername(username: PrismaUser['username'], request: Request) {
```

Comment:
OK, I agree this is the wrong place to do this.  Let's do it in the `root.tsx` loader.

On the server side, we'll read the username/effective username out of session.  On client, we'll send it with the root on the loader.

Reply:
So for client to use it I would make a utils method using useMathcesData() like we have for user?

Reply:
The client would get it from the loader, which can access the `request` and also the db, and put all the data on the User object we return in `root.tsx`.